core/system/rule_manager.go

```diff
@@ -55,9 +31,28 @@ func GetRules() []*SystemRule {
 }
 
 // LoadRules loads given system rules to the rule manager, while all previous rules will be replaced.
-func LoadRules(rules []*SystemRule) (bool, error) {
-	ruleChan <- rules
-	return true, nil
+func LoadRules(rules []*SystemRule) bool {
+	if len(rules) == 0 {
+		logger.Debug("the rules to be loaded is empty")
+		return true
+	}
+
+	if err := onRuleUpdate(rules); err != nil {
+		logger.Errorf("load rules %v failed: %s", rules, err.Error())
+		return false
+	}
+
+	return true
+}
+
+// ClearRules clear all the previous rules
+func ClearRules() bool {
```

Comment:
Could you please provide the use case for function "ClearRules"?

Reply:
In my design, the LoadRules function will discard empty rules, but the original design can set ruleMap to be empty. Therefore, I provided the ClearRules function to do it specially.
For the use case of the function, I haven't thought much about it yet. We can have some discussion about this.